src/execute/hac_sys-pcode-interpreter-calls.adb

```diff
@@ -157,8 +157,11 @@ package body HAC_Sys.PCode.Interpreter.Calls is
           Tasking.Queue (CD, ND, Ident_Index_of_Called, ND.CurTask);  --  put self on entry queue
           Curr_TCB.TS  := WaitRendzv;
           Task_Entered := CD.IdTab (Ident_Index_of_Called).adr_or_sz;  --  Task being entered
-          if ((ND.TCB (Task_Entered).TS = WaitRendzv) and (ND.TCB (Task_Entered).SUSPEND = Ident_Index_of_Called)) or
-             (ND.TCB (Task_Entered).TS = TimedWait)
+          if
+           ((ND.TCB (Task_Entered).TS = WaitRendzv)
+            and then
+            (ND.TCB (Task_Entered).SUSPEND = Ident_Index_of_Called)) or
```

Comment:
Didn't the tool miss here an `or else` opportunity?

Reply:
My suspicion is that the tool (conservatively) assumes that `ND.TCB (Task_Entered).TS` has a side effect,
and hence not always executing it (in case of "or else") will change the behaviour.

If I have time, I will double check!